docs/planning/backend/app/api/fragrances.py

```
@@ -0,0 +1,210 @@
+    response.top_notes = [NoteResponse.model_validate(n) for n in notes_by_position['top']]
+    response.heart_notes = [NoteResponse.model_validate(n) for n in notes_by_position['heart']]
+    response.base_notes = [NoteResponse.model_validate(n) for n in notes_by_position['base']]
+
+    return response
+
+
+@router.post("/lookup", response_model=FragranceResponse)
+def lookup_fragrance(
+    name: str = Query(..., description="Fragrance name to search for"),
+    brand: Optional[str] = Query(None, description="Brand name (helps accuracy)"),
+    db: Session = Depends(get_db)
```

Ruff (0.14.10)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)